frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/advanced-options/LifecycleHookField.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ButtonVariant, Button, Tooltip } from '@patternfly/react-core';
+import { PlusCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { FormikValues, useField, useFormikContext } from 'formik';
+import LifecycleHookForm from './LifecycleHookForm';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import './LifecycleHookField.scss';
+import { getStrategyData } from '../../utils/edit-deployment-utils';
+import { LifecycleAction } from '../utils/types';
+
+const LifecycleHookField: React.FC<{
+  lifecycleHookName: string;
+  resourceObj: K8sResourceKind;
+  dataAttribute: string;
+  title?: string;
+  subTitle?: string;
+}> = ({ title, subTitle, dataAttribute, lifecycleHookName, resourceObj }) => {
+  const { t } = useTranslation();
+  const {
+    setFieldValue,
+    values: {
+      formData: {
+        name: resName,
+        project: { name: resNamespace },
+        deploymentStrategy,
+      },
+    },
+  } = useFormikContext<FormikValues>();
+  const [{ value: lifecycleHookExist }] = useField<boolean>(
+    `formData.deploymentStrategy.lifecycleHooks.${lifecycleHookName}`,
+  );
```

Comment:
can't we get these value from formit context? Why do we need this?

Reply:
Since its a variable depending upon which lch is being added.